client-app/shared/account/composables/useUser.ts

```diff
@@ -87,25 +88,26 @@ export default () => {
     }
   }
 
-  async function registerUser(payload: SignMeUp): Promise<IdentityResultType> {
+  async function registerUser(payload: SignMeUp): Promise<AccountCreationResultType> {
     const { storeId } = globals;
 
     try {
       loading.value = true;
-      const contact = await createContact({
-        firstName: payload.firstName as string,
-        lastName: payload.lastName as string,
-        name: `${payload.firstName} ${payload.lastName}`,
-        emails: [payload.email],
-      });
-      return await createUser({
-        userName: payload.userName,
-        password: payload.password,
-        email: payload.email,
-        memberId: contact.id,
-        userType: "Customer",
+
+      const resultData = await registerAccount({
         storeId,
+        account: {
+          username: payload.userName,
+          password: payload.password,
+          email: payload.email,
+        },
+        contact: {
+          firstName: payload.firstName as string,
```

Comment:
Maybe we should modify payload type to mark required fields to avoid such type conversions?